tests/main/auth-errors/task.yaml

```diff
@@ -5,21 +5,22 @@ systems: [-ubuntu-core-*]
 prepare: |
     mkdir -p /home/test/.snap
     echo -n "{\"macaroon\":\"yummy\",\"discharges\":[ \"some \"]}" > /home/test/.snap/auth.json
+    chown -R test:test /home/test/.snap
 
 restore: |
     rm -rf /home/test/.snap install.output connect.output
 
 execute: |
     echo "An unauthenticated user cannot install snaps"
-    if su - -c "snap install test-snapd-tools 2>${PWD}/install.output" test; then
```

Comment:
Ouch, how did this ever work before?

Reply:
Yeah, I was wondering that too.

Reply:
Can you try to understand that? I think there's something fishy in our test suite still.

Reply:
My bad, I ran su - -c "snap install test-snapd-tools 2>${PWD}/install.output" in spread shell, which ended up creating the file owned by root. Another observation is that although the tests are run by root, /home/gopath is owned by test:test so it all works out in the test user in the end, unless we create files in the test itself and are not careful about changing the permissions. There are more tests that do the su -c ".. ${PWD}>somepath" test pattern, so I'll drop that hunk.

Reply:
Pushed a commit to fix it in other places too. Conveniently, those tests did not execute on Fedora, either by explicit `systems` entry or requiring strict confinement.